src/compiler/checker.ts

```diff
@@ -4022,6 +4029,7 @@ namespace ts {
                     type.typeParameters = concatenate(outerTypeParameters, localTypeParameters);
                     type.outerTypeParameters = outerTypeParameters;
                     type.localTypeParameters = localTypeParameters;
+                    type.minTypeArgumentCount = getMinTypeArgumentCount(localTypeParameters);

```

Comment:
Is there a reason we need the `minTypeArgumentCount` property. Could we just compute it on demand?